libs/api/domains/cms/src/lib/cms.elasticsearch.service.ts

```diff
@@ -175,6 +177,27 @@ export class CmsElasticsearchService {
     return response ? JSON.parse(response) : null
   }
 
+  async getOrganizationSubpage(
+    index: string,
+    { slug, organizationSlug }: GetOrganizationSubpageInput,
+  ): Promise<OrganizationSubpage | null> {
+    // return a single news item by slug
+    const query = {
+      types: ['webOrganizationSubpage'],
+      tags: [
+        { type: 'slug', key: slug },
+        { type: 'organization', key: organizationSlug },
```

Comment:
I understand this as `parentSlug` and `childSlug` ignore if this is not the case.
This approach might not be scalable since you would need one field for each level of the page hierarchy.
We experimented with a custom UI extension called parentSlug to better tackle this type of issue the extension concatenates the parent and child slug allowing for a single non-conflicting field as a page identifier.
This is currently implemented on `aboutPage`(page) and `aboutSubPage`.
We plan to implement this UI extension to articles in a upcoming sprint, It would be useful to get your take on this approach and if it might serve your use case

Reply:
Yes, sounds good. Then we can also start using ReferenceLink instead of hardcoded URLs.

I tested this UI extension some time ago and it didn't work correctly when the slug of the parent was changed. If we can get that working then it's a good enough for us.